2-types/part_1-2/src/part_2.rs

```diff
@@ -0,0 +1,75 @@
+use serde::{Deserialize, Serialize};
+use std::{fs, path::Path};
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct Request {
+    pub r#type: String,
```

Comment:
1. Always avoid `r#` syntax. It is confusing for others
2. You can type-check the request type


```suggestion
#[derive(Serialize, Deserialize, Debug, PartialEq)]
#[serde(rename_all = "lowercase")]
enum RequestType {
    Success,
    Fail,
}
    #[serde(rename = "type")]
    pub request_type: RequestType,
```